x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2035,6 +2035,23 @@ export class RulesClient {
       } catch (e) {
         throw e;
       }
+    }
+
+    let scheduledTaskIdToCreate: string | null = null;
```

Comment:
This logic takes into account the fact that there may be already disabled rules with no corresponding task or that the user may have upgraded from pre-8.0 where the task id was randomly generated and did not match the rule id.

If the rule is disabled with no corresponding task, it's `scheduledTaskId` will be undefined. In this case, we want to schedule a task on enable.

If a rule somehow has a scheduledTaskId defined but it doesn't actually exist, we want to schedule a task on enable.

Finally, if a rule has a scheduledTaskId defined and the task exists, we enable the task.